lib/fennec/evaluator/allocate/request.ex

```diff
@@ -85,12 +85,14 @@ defmodule Fennec.Evaluator.Allocate.Request do
   end
 
   defp verify_unknown_attributes(params, state) do
-    case Params.get_attrs(params) do
-      [] ->
-        {:continue, params, state}
+    with u  <- Params.get_attr(params, Attribute.Username),
+         r  <- Params.get_attr(params, Attribute.Realm),
+         [] <- Params.get_attrs(params) -- [u, r] do
+      {:continue, params, state}
+    else
       _ ->
         {:error, %Attribute.ErrorCode{code: 420}}
```

Comment:
The truth is, that the server shall reject request with this error in case there is at least one `comprehension-required` argument. As far I can see, there arguments have number `0x0000-0x7FFF`. So the correct implementation here would be to check the code of the argument and validate if this code is within `0x0000-0x7FFF` range.